apps/nestjs-api/src/reminder-emails/reminder-emails.service.ts

```diff
@@ -0,0 +1,288 @@
+      Profile_Status__c: ConnectProfileStatus.DRAFTING_PROFILE,
+      CreatedDate: {
+        $gte: jsforce.SfDate.toDateTimeLiteral(createdDate),
+        $lte: jsforce.SfDate.toDateTimeLiteral(endOfCreatedDate),
+      },
+    })
+  }
+
+  async getApprovedMenteesByDaysAndLocation({
+    daysAgo,
+  }: {
+    daysAgo: '7d' | '14d'
+  }) {
+    const approvedDate = new Date()
+    approvedDate.setDate(approvedDate.getDate() - parseInt(daysAgo)) // Interestingly, parseInt is able to parse 7d and 14d to 7 and 14 respectively
+
+    return this.conProfilesService.findAll({
+      'RecordType.DeveloperName': UserType.MENTEE,
+      Profile_Status__c: ConnectProfileStatus.APPROVED,
+      Profile_First_Approved_At__c: {
+        $eq: jsforce.SfDate.toDateLiteral(approvedDate),
+      },
+    })
+  }
+
+  async getConProfilesWithThreeMonthsOldMentorshipMatches() {
+    const NinetyDaysAgo = new Date().setDate(new Date().getDate() - 90)
+
+    const threeMonthsOldMentorshipMatches =
+      await this.conMentorshipMatchesService.findAll({
+        Status__c: MentorshipMatchStatus.ACCEPTED,
+        Start_Date__c: {
+          $eq: jsforce.SfDate.toDateLiteral(NinetyDaysAgo),
+        },
+      })
+
+    const menteeIds = threeMonthsOldMentorshipMatches.map(
+      (match) => match.props.menteeId
+    )
+    const mentorIds = threeMonthsOldMentorshipMatches.map(
+      (match) => match.props.mentorId
+    )
+
+    if ([...menteeIds, ...mentorIds].length > 0) {
+      return await this.conProfilesService.findAll({
+        'Contact__r.Id': {
+          $in: [...menteeIds, ...mentorIds],
+        },
+      })
+    }
+
+    return []
+  }
+
+  async sendCompleteProfileReminder({
+    userType,
+    email,
+    firstName,
+  }: {
+    userType: UserType
+    email: string
+    firstName: string
+  }) {
+    if (!userType) {
+      throw new Error('User type is required')
+    }
+
+    const sfEmailTemplateName =
+      userType === UserType.MENTOR
+        ? 'Mentor - Profile Completion Reminder'
+        : userType === UserType.MENTEE
+        ? 'Mentee - Profile Completion Reminder'
+        : null
+
+    if (!sfEmailTemplateName) {
+      throw new Error('Invalid user type. Must be either MENTOR or MENTEE')
+    }
+
+    const template = await this.emailTemplatesService.getEmailTemplate(
+      sfEmailTemplateName
+    )
+
+    if (!template) {
+      throw new Error(`Email template not found: ${sfEmailTemplateName}`)
+    }
+
+    const sanitizedHtml = template.HtmlValue.replace(
+      /{{{Recipient\.FirstName}}}/g,
+      `${firstName} ${email}`
+    )
+
+    const params = {
+      Destination: {
+        ToAddresses: isProductionOrDemonstration()
+          ? ['anilakarsu93@gmail.com']
+          : [process.env.NX_DEV_MODE_EMAIL_RECIPIENT],
+      },
+      Message: {
+        Body: {
+          Html: { Data: sanitizedHtml },
+        },
+        Subject: { Data: template.Subject },
+      },
+      Source: 'career@redi-school.org',
```

Comment:
Will we get into trouble with Malmö?

Reply:
Again, I can do the fine tuning once I see everything is working correctly.